azuredevops/internal/service/serviceendpoint/data_serviceendpoint_generic_v2.go

```diff
@@ -0,0 +1,194 @@
+package serviceendpoint
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	SE "github.com/microsoft/azure-devops-go-api/azuredevops/v7/serviceendpoint"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/client"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils"
+)
+
+// DataServiceEndpointGenericV2 schema and implementation for generic service endpoint data source
+func DataServiceEndpointGenericV2() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataServiceEndpointGenericV2Read,
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(1 * time.Minute),
+		},
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.IsUUID,
+			},
+			"shared_project_ids": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			// Use id to look up by ID
+			"id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.IsUUID,
+				ExactlyOneOf: []string{"id", "name"},
+			},
+			// Use name to look up by name
+			"name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotWhiteSpace,
+				ExactlyOneOf: []string{"id", "name"},
```

Comment:
These two attributes need to be Optional+Computed, so that when "A" is specified, "B" is still exposed to be used by the user.